src/IB/IIMethod.cpp

```diff
@@ -762,6 +810,7 @@ IIMethod::interpolateVelocity(const int u_data_idx,
             U_t_vec = d_U_t_new_vecs[part];
             X_vec = d_X_new_vecs[part];
         }
+        
         copy_and_synch(*X_vec, *X_ghost_vec);
```

Comment:
We already do the time-based selection of the correct vectors here. Could you use these in `batch_vec_ghost_update()` instead?

If vectors are used with patch data then we need to ensure that they are IB-ghosted: normally `FEDataManager` handles that but its all done manually here. Which of these need the IB ghosting? I'm surprised that only the position and velocity jump are explicitly labeled as such.